### PR TITLE
Fix multithread tests

### PR DESCRIPTION
**Pull request by dustinblack**
## Changes introduced with this PR

~~When coremark runs `make` in a python `subprocess`, it seems to be not allowed to spawn the multiple testing threads needed when multiple contexts are used, resulting in invalid results. This fix introduces a change in the way the `subprocess` is called, allowing for the required multithreaded behavior.~~

The `make` command called by `subprocess` to run the coremark-pro benchmarks uses an `XCMD=` set of sub-parameters that are passed on to the individual benchmarks. It turns out that quoting the string passed to `XCMD=` causes a parsing problem when those sub-parameters are passed to the benchmarks. This change removes the quote marks from the string.

Also adds the command line to the output, and improves a regex use to remove a deprecation warning message.

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: arcaflow_plugin_coremark_pro/coremark_pro_plugin.py [hunk 1]
```diff
@@ -136,8 +142,9 @@ def certify_all(
     }
 
     # Construct the output object
+    exclusion_search = re.compile(r"^(\s+|Starting|Workload|-|Mark)", re.IGNORECASE)
     for line in ca_return[1].splitlines():
-        if re.match(r'^(\s+|Starting|W(?i)[orkload]|-|M(?i)[ark])', line):
+        if exclusion_search.match(line):
```
Comment (webbnh) [inline arcaflow_plugin_coremark_pro/coremark_pro_plugin.py:146]: These two regex's don't look remotely equivalent, but I think that's good, because the old one looks "broken" to me.  🙂

**Review by webbnh — APPROVED** [review]
The changes look fine to me, although I don't see why or how injecting the shell into the execution would make any difference.

File: arcaflow_plugin_coremark_pro/coremark_pro_plugin.py [hunk 2]
```diff
@@ -16,10 +16,14 @@
...
-def run_oneshot_cmd(command_list, workdir) -> str:
+def run_oneshot_cmd(command, workdir) -> str:
     try:
         cmd_out = subprocess.check_output(
-            command_list, stderr=subprocess.STDOUT, text=True, cwd=workdir
+            command,
+            stderr=subprocess.STDOUT,
+            text=True,
+            cwd=workdir,
+            shell=True,
```
Comment (mfleader) [inline arcaflow_plugin_coremark_pro/coremark_pro_plugin.py]: Can you add a comment about why `shell=True` is necessary?

**Review by mfleader — CHANGES_REQUESTED** [review]

File: arcaflow_plugin_coremark_pro/coremark_pro_plugin.py (hunk 2 above)
Comment (dustinblack) [reply arcaflow_plugin_coremark_pro/coremark_pro_plugin.py]: We found a better approach, so now it is not needed.

File: arcaflow_plugin_coremark_pro/coremark_pro_plugin.py [hunk 3]
```diff
@@ -16,10 +16,13 @@
...
-def run_oneshot_cmd(command_list, workdir) -> str:
+def run_oneshot_cmd(command, workdir) -> str:
     try:
         cmd_out = subprocess.check_output(
-            command_list, stderr=subprocess.STDOUT, text=True, cwd=workdir
+            command,
+            stderr=subprocess.STDOUT,
+            text=True,
+            cwd=workdir,
         )
```
Comment (webbnh) [inline arcaflow_plugin_coremark_pro/coremark_pro_plugin.py]: You don't want to revert this change?

**Review by webbnh — APPROVED** [review]
Looks good...just one nit.

File: arcaflow_plugin_coremark_pro/coremark_pro_plugin.py (hunk 1 above)
Comment (dustinblack) [reply arcaflow_plugin_coremark_pro/coremark_pro_plugin.py:146]: The `(?i)`'s were intended to ignore the case of the substring, but python kept barking about deprecation. So the `re.IGNORECASE` in the new `re.compile` command just ignores the case altogether, which turns out to be fine for this query.

**Comment by webbnh** [comment]
You should also update the PR description with our "learnings".

File: arcaflow_plugin_coremark_pro/coremark_pro_plugin.py (hunk 3 above)
Comment (dustinblack) [reply arcaflow_plugin_coremark_pro/coremark_pro_plugin.py]: It just doesn't matter much.

Comment (webbnh) [reply arcaflow_plugin_coremark_pro/coremark_pro_plugin.py]: This change has zero impact on the code's execution but it will show up in a future retrospection or forensic search, and it would be best to avoid that sort of "false positive" (and, if you don't commit this change, then it cannot be the basis for a "positive" which is _not_ "false").

File: arcaflow_plugin_coremark_pro/coremark_pro_plugin.py [hunk 4]
```diff
@@ -114,7 +117,7 @@ def certify_all(
         xcmd.append(f"-c{params.contexts}")
     if params.workers:
         xcmd.append(f"-w{params.workers}")
-    ca_cmd.append(f"XCMD={' '.join(xcmd)!r}")
+    ca_cmd.append(f"XCMD={' '.join(xcmd)}")
```
Comment (mfleader) [inline arcaflow_plugin_coremark_pro/coremark_pro_plugin.py:122]: Can you add this part (or a paraphrase of it) of your pull request description here (or near here) to explain why this is done?

>>> The make command called by subprocess to run the coremark-pro benchmarks uses an XCMD= set of sub-parameters that are passed on to the individual benchmarks. It turns out that quoting the string passed to XCMD= causes a parsing problem when those sub-parameters are passed to the benchmarks.

**Review by mfleader — CHANGES_REQUESTED** [review]

File: arcaflow_plugin_coremark_pro/coremark_pro_plugin.py (hunk 4 above)
Comment (dustinblack) [reply arcaflow_plugin_coremark_pro/coremark_pro_plugin.py:122]: I'm not sure what you are asking for. Do you want me to explain how the `!r` operative worked before it was removed from the code?

Comment (mfleader) [reply arcaflow_plugin_coremark_pro/coremark_pro_plugin.py:122]: Actually, I think I Just need your short explanation of why the `XCMD` string is created as a code comment at this point in the source code.

> coremark-pro benchmarks use an `XCMD=` set of sub-parameters that are passed on to the individual benchmarks

**Review by mfleader — APPROVED** [review]

**Review by webbnh — APPROVED** [review]
👍
